core/common/src/Utf8.kt

```diff
@@ -122,14 +122,23 @@ internal fun String.utf8Size(startIndex: Int = 0, endIndex: Int = length): Long
 /**
  * Encodes [codePoint] in UTF-8 and writes it to this sink.
  *
+ * Note that in general, a value retrieved from [Char.code] could not be written directly
+ * as it may be a part of a [surrogate pair](https://www.unicode.org/faq/utf_bom.html#utf16-2) (that could be
+ * detected using [Char.isSurrogate], or [Char.isHighSurrogate] and [Char.isLowSurrogate]).
+ * Such a pair of characters needs to be manually converted back to a single code point
+ * which then could be written to a [Sink].
+ * Without such a conversion, data written to a [Sink] can not be converted back
+ * to a string from which a surrogate pair was retrieved.
+ *
```

Comment:
The documentation does not say anything about what will be written if we pass a code point with the surrogate value (`U+d800`..`U+dfff`).
E.g. on the JVM will be written single byte with value 63 (`?`)

Also we should add test on it

Reply:
Currently, we always writing '?'. `Utf8Test::writeSurrogateCodePoint` covers that scenario.

Reply:
But this is not mentioned in the KDoc

Reply:
Thanks for pointing to that! Opened https://github.com/Kotlin/kotlinx-io/pull/314